### ISSUE TITLE
github attached videos

### ISSUE THREAD

**Issue by illume** [remarkjs/react-markdown#848]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

Github adds a link in the text when you upload a video like this:

`https://github.com/user-attachments/assets/46b377f2-a949-4e6e-b82a-bf528e7fd42a`

There's no markup around it, it just leave that text link when it's uploaded. When it renders, it renders as a html5 video player.

### Solution

Find github user-attachments and render them with a player.

### Alternatives

I guess this is a very github specific thing. If there's some other way to render videos it might be nice to find a work around for people. Perhaps pre-processing and generating the markup somehow?

**Comment by illume** [comment]
I could help make this if someone can point me in the right direction. I can also work on a workaround for this if this isn't wanted inside react-markdown itself.

**Comment by ChristianMurphy** [comment]
Welcome @illume! 👋 

TL;DR this doesn't belong in `react-markdown` core, but is possible.

---

`react-markdown` provided CommonMark out of the box. https://commonmark.org/
This can be extended with GitHub Flavored Markdown https://github.github.com/gfm/ (https://github.com/remarkjs/remark-gfm)

The video player feature is not part of either standard 😅 
GitHub the UI adds that on top of both standards.
GitHub includes a bunch of custom players https://github.com/orgs/remarkjs/discussions/932#discussioncomment-3437350

This would probably live as a rehype plugin https://unifiedjs.com/learn/
or as a custom `a` component to use the player you want to leverage https://github.com/remarkjs/react-markdown?tab=readme-ov-file#components

**Comment by illume** [comment]
Thank you for the pointers :)
